### ISSUE TITLE
Error installing numpy on pyston virtualenv

### ISSUE THREAD

**Issue by amirza1** [dropbox/pyston#1344]
I am trying to install pyston+numpy as per the instructions on your main project page on Ubuntu 14.04
(I am probably missing some steps/prerequisites)

I have:
1. downloaded and extracted pyston-0.5.1-linux64.tar.gz to ~/pyston-0.5.1-linux64, then run:
2. `~/pyston-0.5.1-linux64/pyston ~/pyston-0.5.1-linux64/virtualenv/virtualenv.py ~/pystonenv`
3. `source ~/pystonenv/bin/activate`
4. `pip install https://github.com/dropbox/pyston/releases/download/v0.5.1/Cython-0.24-pyston.tar.gz`
5. `pip install git+git://github.com/numpy/numpy@v1.11.0`

This yielded the following output:
```
Collecting git+git://github.com/numpy/numpy@v1.11.0
  Cloning git://github.com/numpy/numpy (to v1.11.0) to /tmp/pip-e8SCl4-build
Installing collected packages: numpy
  Running setup.py install for numpy
    Running from numpy source directory.
    Processing numpy/random/mtrand/mtrand.pyx
    Note: if you need reliable uninstall behavior, then install
    with pip instead of using `setup.py install`:
      - `pip install .`       (from a git repo or downloaded source
                               release)
      - `pip install numpy`   (last Numpy release on PyPi)
    Cythonizing sources
    blas_opt_info:
    blas_mkl_info:
      libraries mkl,vml,guide not found in ['/usr/local/lib', '/usr/lib', '/usr/lib/x86_64-linux-gnu']
      NOT AVAILABLE
    openblas_info:
      libraries openblas not found in ['', '']
    Runtime library openblas was not found. Ignoring
      FOUND:
        libraries = ['openblas', 'openblas']
        library_dirs = ['/usr/lib']
        language = c
        define_macros = [('HAVE_CBLAS', None)]
      FOUND:
        libraries = ['openblas', 'openblas']
        library_dirs = ['/usr/lib']
        language = c
        define_macros = [('HAVE_CBLAS', None)]
    non-existing path in 'numpy/distutils': 'site.cfg'
    F2PY Version 2
    lapack_opt_info:
    openblas_lapack_info:
    C compiler: cc
    compile options: '-c -Werror=implicit-function-declaration'
    cc: /tmp/tmpMd8lad/source.c
    /tmp/tmpMd8lad/source.c: In function ‘main’:
    /tmp/tmpMd8lad/source.c:4:13: error: implicit declaration of function ‘zungqr_’ [-Werror=implicit-function-declaration]
                 zungqr_();
                 ^
    cc1: some warnings being treated as errors
    /tmp/tmpMd8lad/source.c: In function ‘main’:
    /tmp/tmpMd8lad/source.c:4:13: error: implicit declaration of function ‘zungqr_’ [-Werror=implicit-function-declaration]
                 zungqr_();
                 ^
    cc1: some warnings being treated as errors
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-e8SCl4-build/setup.py", line 386, in <module>
        setup_package()
      File "/tmp/pip-e8SCl4-build/setup.py", line 378, in setup_package
        setup(**metadata)
      File "/tmp/pip-e8SCl4-build/numpy/distutils/core.py", line 135, in setup
        config = configuration()
      File "/tmp/pip-e8SCl4-build/setup.py", line 161, in configuration
        config.add_subpackage('numpy')
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 1001, in add_subpackage
        config_list = self.get_subpackage(subpackage_name, subpackage_path,
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 967, in get_subpackage
        config = self._get_configuration_from_setup_py(
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 909, in _get_configuration_from_setup_py
        config = setup_module.configuration(*args)
      File "numpy/setup.py", line 16, in configuration
        config.add_subpackage('linalg')
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 1001, in add_subpackage
        config_list = self.get_subpackage(subpackage_name, subpackage_path,
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 967, in get_subpackage
        config = self._get_configuration_from_setup_py(
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 909, in _get_configuration_from_setup_py
        config = setup_module.configuration(*args)
      File "numpy/linalg/setup.py", line 26, in configuration
        lapack_info = get_info('lapack_opt', 0)  # and {}
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 372, in get_info
        return cl().get_info(notfound_action)
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 560, in get_info
        self.calc_info()
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 1482, in calc_info
        openblas_info = get_info('openblas_lapack')
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 372, in get_info
        return cl().get_info(notfound_action)
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 560, in get_info
        self.calc_info()
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 1754, in calc_info
        if not self.check_embedded_lapack(info):
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 1788, in check_embedded_lapack
        obj = c.compile([src], output_dir=tmpdir)
      File "/tmp/pip-e8SCl4-build/numpy/distutils/ccompiler.py", line 230, in CCompiler_compile
        single_compile(o)
      File "/tmp/pip-e8SCl4-build/numpy/distutils/ccompiler.py", line 194, in single_compile
        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
      File "/tmp/pip-e8SCl4-build/numpy/distutils/unixccompiler.py", line 52, in UnixCCompiler__compile
        raise CompileError(msg)
    distutils.errors.CompileError: Command "cc -c -Werror=implicit-function-declaration /tmp/tmpMd8lad/source.c -o /tmp/tmpMd8lad/tmp/tmpMd8lad/source.o" failed with exit status 1
    Complete output from command /home/sight/new/pystonenv/bin/pyston -c "import setuptools, tokenize;__file__='/tmp/pip-e8SCl4-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Q9AH8h-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/sight/new/pystonenv/include/site/python2.7:
    Running from numpy source directory.
    
    Processing numpy/random/mtrand/mtrand.pyx
    
    
    
    Note: if you need reliable uninstall behavior, then install
    
    with pip instead of using `setup.py install`:
    
    
    
      - `pip install .`       (from a git repo or downloaded source
    
                               release)
    
      - `pip install numpy`   (last Numpy release on PyPi)
    
    
    
    
    
    Cythonizing sources
    
    blas_opt_info:
    
    blas_mkl_info:
    
      libraries mkl,vml,guide not found in ['/usr/local/lib', '/usr/lib', '/usr/lib/x86_64-linux-gnu']
    
      NOT AVAILABLE
    
    
    
    openblas_info:
    
      libraries openblas not found in ['', '']
    
    Runtime library openblas was not found. Ignoring
    
      FOUND:
    
        libraries = ['openblas', 'openblas']
    
        library_dirs = ['/usr/lib']
    
        language = c
    
        define_macros = [('HAVE_CBLAS', None)]
    
    
    
      FOUND:
    
        libraries = ['openblas', 'openblas']
    
        library_dirs = ['/usr/lib']
    
        language = c
    
        define_macros = [('HAVE_CBLAS', None)]
    
    
    
    non-existing path in 'numpy/distutils': 'site.cfg'
    
    F2PY Version 2
    
    lapack_opt_info:
    
    openblas_lapack_info:
    
    C compiler: cc
    
    
    
    creating /tmp/tmpMd8lad/tmp
    
    creating /tmp/tmpMd8lad/tmp/tmpMd8lad
    
    compile options: '-c -Werror=implicit-function-declaration'
    
    cc: /tmp/tmpMd8lad/source.c
    
    /tmp/tmpMd8lad/source.c: In function ‘main’:
    
    /tmp/tmpMd8lad/source.c:4:13: error: implicit declaration of function ‘zungqr_’ [-Werror=implicit-function-declaration]
    
                 zungqr_();
    
                 ^
    
    cc1: some warnings being treated as errors
    
    /tmp/tmpMd8lad/source.c: In function ‘main’:
    
    /tmp/tmpMd8lad/source.c:4:13: error: implicit declaration of function ‘zungqr_’ [-Werror=implicit-function-declaration]
    
                 zungqr_();
    
                 ^
    
    cc1: some warnings being treated as errors
    
    Traceback (most recent call last):
    
      File "<string>", line 1, in <module>
    
      File "/tmp/pip-e8SCl4-build/setup.py", line 386, in <module>
    
        setup_package()
    
      File "/tmp/pip-e8SCl4-build/setup.py", line 378, in setup_package
    
        setup(**metadata)
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/core.py", line 135, in setup
    
        config = configuration()
    
      File "/tmp/pip-e8SCl4-build/setup.py", line 161, in configuration
    
        config.add_subpackage('numpy')
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 1001, in add_subpackage
    
        config_list = self.get_subpackage(subpackage_name, subpackage_path,
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 967, in get_subpackage
    
        config = self._get_configuration_from_setup_py(
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 909, in _get_configuration_from_setup_py
    
        config = setup_module.configuration(*args)
    
      File "numpy/setup.py", line 16, in configuration
    
        config.add_subpackage('linalg')
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 1001, in add_subpackage
    
        config_list = self.get_subpackage(subpackage_name, subpackage_path,
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 967, in get_subpackage
    
        config = self._get_configuration_from_setup_py(
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/misc_util.py", line 909, in _get_configuration_from_setup_py
    
        config = setup_module.configuration(*args)
    
      File "numpy/linalg/setup.py", line 26, in configuration
    
        lapack_info = get_info('lapack_opt', 0)  # and {}
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 372, in get_info
    
        return cl().get_info(notfound_action)
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 560, in get_info
    
        self.calc_info()
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 1482, in calc_info
    
        openblas_info = get_info('openblas_lapack')
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 372, in get_info
    
        return cl().get_info(notfound_action)
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 560, in get_info
    
        self.calc_info()
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 1754, in calc_info
    
        if not self.check_embedded_lapack(info):
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/system_info.py", line 1788, in check_embedded_lapack
    
        obj = c.compile([src], output_dir=tmpdir)
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/ccompiler.py", line 230, in CCompiler_compile
    
        single_compile(o)
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/ccompiler.py", line 194, in single_compile
    
        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
    
      File "/tmp/pip-e8SCl4-build/numpy/distutils/unixccompiler.py", line 52, in UnixCCompiler__compile
    
        raise CompileError(msg)
    
    distutils.errors.CompileError: Command "cc -c -Werror=implicit-function-declaration /tmp/tmpMd8lad/source.c -o /tmp/tmpMd8lad/tmp/tmpMd8lad/source.o" failed with exit status 1
    
    ----------------------------------------
    Command "/home/sight/new/pystonenv/bin/pyston -c "import setuptools, tokenize;__file__='/tmp/pip-e8SCl4-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Q9AH8h-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/sight/new/pystonenv/include/site/python2.7" failed with error code 1 in /tmp/pip-e8SCl4-build


```

**Comment by undingen** [comment]
We pass ` -Werror=implicit-function-declaration` per default as compiler flag to extension during the compilation which causes problems for you :-(.
I think you could try to remove the `cc_args = cc_args + ["-Werror=implicit-function-declaration"]` in `from_cpython/Lib/distutils/ccompiler.py` (mmh we run into this now several times we should probably remove it...)
I did not run into this problem because I used numpy with atlas. So maybe installing `libatlas-base-dev` will fix the issue without having to change the compiler flags.
Btw: there is also a docker image which contains pyston and a build numpy ` docker run -it pyston/pyston-numpy`

**Comment by kmod** [comment]
Ok this should be fixed in master via #1345, but until we do another release you should be able to do use of the workarounds that undingen mentioned.  Thanks for the report!
